installer

```diff
@@ -0,0 +1,139 @@
+#!/usr/local/cpython-3.4/bin/python3
+
+"""
+Copy modules from here to --prefix.
+
+Note that we do not scan for directories; instead we have a hardcoded list in all_directories.
+"""
+
+import os
+import shutil
+import sys
+
+
+prefix = os.path.expanduser('~/.micropython/lib')
+directories = []
+all_directories = ['micropython', 'python-ecosys', 'python-stdlib', 'tools', 'unix-ffi']
+
+
+def usage(retval):
+    """Output a usage message."""
+    if retval == 0:
+        write = sys.stdout.write
+    else:
+        write = sys.stderr.write
+
+    write('Usage: {} --prefix ~/.micropython/lib --directories {}\n'.format(sys.argv[0], ' '.join(all_directories)))
+    write('--directories, if provided, must be the last argument specified\n')
+
+    sys.exit(retval)
+
+
+# The getopt-like module du jour would be a little more concise, but that's much harder for a static analyzer to check.
+while sys.argv[1:]:
+    if sys.argv[1] == '--prefix':
+        prefix = os.path.expanduser(sys.argv[2])
+        del sys.argv[1]
+    elif sys.argv[1] == '--directories':
+        directories = sys.argv[2:]
+        break
+    elif sys.argv[1] in ('-h', '--help'):
+        usage(0)
+    else:
+        sys.stderr.write('{}: unrecognized option: {}\n'.format(sys.argv[0], sys.argv[1]))
+        usage(1)
+    del sys.argv[1]
+
+if directories == []:
+    directories = all_directories
+
+
+class Error(OSError):
+    """An exception to raise when an error is encountered copying."""
+
+    pass
+
+
+def copytree(src, dst, symlinks=False, ignore=None, copy_function=shutil.copy2,
+             ignore_dangling_symlinks=False):
+    """
+    Recursively copy a directory tree.
+
+    This is a slightly-modified copy of copytree from CPython 3.4.  The motivation was to allow dst to preexist.
```

Comment:
Surely there are simpler ways, no? Doesn't seem right to introduce dependency on such a large function simply to copy some files

Reply:
It's not really a dependency, but this is a library function that lacks a dirs_exist_ok option.  That's not present until Python 3.8+ though.

Reply:
Ok. But it's a dependncy on a lot of non-trivial code unrelated to the actual task at hand. Plus I'm not sure one can just copy from CPython without mentioning its license? Anyway, dirs_exist_ok is actaully a problem in my experience: what typcially happens is a file gets deleted from the source (this repository) but then it just remains in the target and causes havoc there. Deleting all directories first (as in: loop over the first level and delete everything encounterd in prefix) is better and safer. Plus makes copy_tree usable again.

Reply:
The code can be trivially removed when 3.8 is adopted.

CPython is fully opensource.

Goodness, no, deleting a hierarchy isn't even close to safer.

Reply:
> Goodness, no, deleting a hierarchy isn't even close to safer.

How so (in this particular case)? Do you know another way to deal with the problem I mentioned?